homeassistant/components/ozw/websocket_api.py

```diff
@@ -102,6 +114,96 @@ def websocket_get_nodes(hass, connection, msg):
     )
 
 
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "ozw/get_config_parameters",
+        vol.Required(NODE_ID): vol.Coerce(int),
+        vol.Optional(OZW_INSTANCE, default=1): vol.Coerce(int),
+    }
+)
+def websocket_get_config_parameters(hass, connection, msg):
+    """Get a list of configuration parameters for an OZW node instance."""
+    try:
+        node = get_node_from_manager(
+            hass.data[DOMAIN][MANAGER], msg[OZW_INSTANCE], msg[NODE_ID]
+        )
+    except NotFoundError as err:
+        connection.send_error(
+            msg[ID],
+            websocket_api.const.ERR_NOT_FOUND,
+            err.args[0],
+        )
+        return
+
+    connection.send_result(
+        msg[ID],
+        get_config_parameters(node),
+    )
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "ozw/set_config_parameter",
+        vol.Required(NODE_ID): vol.Coerce(int),
+        vol.Optional(OZW_INSTANCE, default=1): vol.Coerce(int),
+        vol.Required(PARAMETER): vol.Coerce(int),
+        vol.Required(VALUE): vol.Any(
+            vol.All(
+                cv.ensure_list,
+                [
+                    vol.All(
+                        {
+                            vol.Exclusive(ATTR_LABEL, "bit"): cv.string,
+                            vol.Exclusive(ATTR_POSITION, "bit"): vol.Coerce(int),
+                            vol.Required(ATTR_VALUE): bool,
+                        },
+                        cv.has_at_least_one_key(ATTR_LABEL, ATTR_POSITION),
+                    )
+                ],
+            ),
+            vol.Coerce(int),
+            bool,
+            cv.string,
+        ),
+    }
+)
+def websocket_set_config_parameter(hass, connection, msg):
+    """Set a config parameter to a node."""
+    try:
+        node = get_node_from_manager(
+            hass.data[DOMAIN][MANAGER], msg[OZW_INSTANCE], msg[NODE_ID]
+        )
+    except NotFoundError as err:
+        connection.send_error(
+            msg[ID],
+            websocket_api.const.ERR_NOT_FOUND,
+            err.args[0],
+        )
+        return
+
+    try:
+        set_config_parameter(
+            node,
+            msg[PARAMETER],
+            msg[VALUE],
+        )
+        connection.send_result(msg[ID])
```

Comment:
Please move `connection.send_result` to the bottom out of the try... except block.

Reply:
Why? The result should only be sent if the call succeeds - if it fails, then we send an error message. By moving the `send_result` call out of the try except block it would always get sent

Reply:
We only want to wrap the line that can raise in the try... except block. We return in the except block so it won't happen unless the call is successful.

Reply:
good point, thanks for the explanation